pilot/pkg/model/jwks_resolver.go

```diff
@@ -384,41 +394,53 @@ func (r *JwksResolver) refresher() {
 	for {
 		select {
 		case <-r.refreshTicker.C:
-			currentHasError := r.refresh()
-			if currentHasError {
-				if lastHasError {
-					// update to exponential backoff if last time also failed.
-					r.refreshInterval *= 2
-					if r.refreshInterval > JwtPubKeyRefreshIntervalOnFailureResetThreshold {
-						r.refreshInterval = JwtPubKeyRefreshIntervalOnFailureResetThreshold
-					}
-				} else {
-					// change to the refreshIntervalOnFailure if failed for the first time.
-					r.refreshInterval = r.refreshIntervalOnFailure
-				}
-			} else {
-				// reset the refresh interval if success.
-				r.refreshInterval = r.refreshDefaultInterval
+			if !r.jwksUribackgroundChannel {
+				lastHasError = r.refreshCache(lastHasError)
 			}
-			lastHasError = currentHasError
-			r.refreshTicker.Reset(r.refreshInterval)
 		case <-closeChan:
 			r.refreshTicker.Stop()
 			return
+		case <-jwksuriChannel:
+			r.jwksUribackgroundChannel = true
```

Comment:
style: maybe we can just make `jwksUribackgroundChannel` a parameter to `refreshCache`? Using the struct level variable makes it pretty hard to follow

Reply:
With the struct level variable we can stop the main ticker execution when our background execution is going on. This would resolve the consistency issues, added a check on line no. 397